src/v3/providers/coingecko/coingecko.ts

```diff
@@ -127,6 +132,67 @@ automatedTokenMappingsSyncDoc.onChange(automatedMappings => {
   }
 })
 
+const coingeckoToCrossChainMapping = async (
+  coingeckoAssets: ReturnType<typeof asCoingeckoAssetResponse>,
+  tokenTypes: StringNullMap
```

Comment:
tokenTypes should be more strongly typed. Instead of a StringNullMap, change to 

type TokenType = 'simple' | 'evm' | 'cosmos' | 'xrpl' | 'colon-delimited' | 'lowercase'

type TokenTypeMap = Record<string, TokenType>